observer/probers/tls/tls.go

```diff
@@ -92,102 +84,80 @@ func getOCSPStatus(cert, issuer *x509.Certificate) (int, error) {
 		return ocsp.Unknown, err
 	}
 
-	switch ocspRes.Status {
-	case ocsp.Revoked:
-		return ocsp.Revoked, nil
-	case ocsp.Good:
-		return ocsp.Good, nil
-	default:
-		return ocsp.Unknown, nil
-	}
+	return ocspRes.Status, nil
 }
 
-// Make a TLS connection with InsecureSkipVerify set to true to get the
-// expiration time and root information of insecure leaf certificates.
-func (p TLSProbe) getInsecureCertInfo() (string, string, time.Duration, error) {
-	conf := &tls.Config{InsecureSkipVerify: true}
-	conn, err := tls.Dial("tcp", p.url+":443", conf)
+// Export expiration timestamp and reason to Prometheus.
+func (p TLSProbe) exportMetrics(notAfter time.Time, reason reason) {
+	p.notAfter.WithLabelValues(p.url).Set(float64(notAfter.Unix()))
+	p.notAfter.WithLabelValues(p.url, reasonToString[reason]).Inc()
+}
+
+func (p TLSProbe) probeExpired(timeout time.Duration) (bool, time.Duration) {
+	start := time.Now()
+	conn, err := tls.DialWithDialer(&net.Dialer{Timeout: timeout}, "tcp", p.url+":443", &tls.Config{InsecureSkipVerify: true})
 	if err != nil {
-		return "", "", 0, err
+		p.exportMetrics(time.Time{}, internalError)
+		return false, time.Since(start)
 	}
+	defer conn.Close()
 	peers := conn.ConnectionState().PeerCertificates
-	expiryDate := peers[0].NotAfter
-
 	for i := 0; i < len(peers)-1; i++ {
 		certIssuer, issuerSubject := peers[i].Issuer, peers[i+1].Subject
 		if (certIssuer.CommonName != issuerSubject.CommonName) || (certIssuer.Organization[0] != issuerSubject.Organization[0]) {
-			err = fmt.Errorf(reasonToString[issuerVerifyFailed])
+			p.exportMetrics(peers[0].NotAfter, issuerVerifyFailed)
+			return false, time.Since(start)
 		}
 	}
-
-	root := peers[len(peers)-1].Issuer
-	return root.Organization[0], root.CommonName, time.Until(expiryDate), err
-}
-
-// Check if the root of the leaf's certificate is the one we expect.
-func (p TLSProbe) isRootExpected(rootO, rootCN string) bool {
-	if p.rootOrg == rootO && p.rootCN == rootCN {
-		return true
+	p.exportMetrics(peers[0].NotAfter, none)
+	if time.Until(peers[0].NotAfter) > 0 {
+		return false, time.Since(start)
 	}
-	return false
-}
+	root := peers[len(peers)-1].Issuer
+	return p.rootOrg == root.Organization[0] && p.rootCN == root.CommonName, time.Since(start)
```

Comment:
This is super minor, but we'd usually reverse these conditionals: `root.Organization[0] == p.rootOrg && root.CommonName == p.rootCN`.

This is because we usually construct equality tests as `if actual == expected`. By analogy, we don't say `if 0 == myCounter`, we say `if myCounter == 0`.

If you want to go a step further in readability and consistency, you could format this exactly like you do on line 107:
```
if root.Organization[0] != p.rootOrg || root.CommonName != p.rootCN {
  return false, time.Since(start)
}
return true, time.Since(start)
```
Yes, it's more verbose, but consistent structure helps readers come to conclusions about correctness quicker.